src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -12,100 +11,179 @@
+            long ddlHandlerTimeoutSec = 60L;
+            if (!KitUtils.isEmpty(ddlTimeout)) {
+                try {
+                    ddlHandlerTimeoutSec = Long.parseLong(ddlTimeout);
+                } catch (Exception unableToGetDDLTimeout) {
+                }
+            }
+
+            SingularConfig config = new SingularConfig(singularKey, singularSecret);
+            config.withDDLTimeoutInSec(ddlHandlerTimeoutSec);
+
+            Activity activity = getCurrentActivity().get();
+
+            if (activity != null) {
+                Intent intent = activity.getIntent();
+
+                config.withSingularLink(intent, new SingularLinkHandler() {
+                    @Override
+                    public void onResolved(SingularLinkParams singularLinkParams) {
+                        AttributionResult attributionResult = new AttributionResult();
+                        attributionResult.setServiceProviderId(MParticle.ServiceProviders.SINGULAR);
+                        attributionResult.setLink(singularLinkParams.getDeeplink());
+                        try {
+                            JSONObject linkParams = new JSONObject();
+                            linkParams.put(PASSTHROUGH, singularLinkParams.getPassthrough());
+                            linkParams.put(IS_DEFERRED, singularLinkParams.isDeferred());
+                            attributionResult.setParameters(linkParams);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+                        getKitManager().onResult(attributionResult);
+                    }
+                });
+            }
+
+            // If the environment is in development mode, enable logging.
+            if (MPUtility.isDevEnv()) {
+                config.withLoggingEnabled();
+                config.withLogLevel(Log.DEBUG);
+            }
+
+            Singular.setWrapperNameAndVersion(MPARTICLE_WRAPPER_NAME, MPARTICLE_WRAPPER_VERSION);
+
+            return config;
+
+        } catch (Exception ex) {
+            logger.error("Can't build Singular Config in the mParticle Kit", ex);
+            return null;
+        }
+    }
+
     @Override
     public List<ReportingMessage> setOptOut(boolean b) {
         return null;
     }
 
     @Override
     public String getName() {
-        return "Singular";
+        return KIT_NAME;
     }
 
     @Override
-    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
-        return null;
+    public void setInstallReferrer(Intent intent) {
     }
 
-    @Override
-    public List<ReportingMessage> onActivityStarted(Activity activity) {
-        return null;
-    }
+    //endregion
+
+    //region Activity Listener Implementation
 
     @Override
     public List<ReportingMessage> onActivityResumed(Activity activity) {
         Singular.onActivityResumed();
+        // TODO: Check the meaning of returning null here
```

Comment:
`null` is a perfectly acceptable response here. 

`ReportingMessage` are used to track any "Event" data (MPEvent, CommerceEvent, PushRegistration, etc) that is forwarded to the underlying SDK, in your case the Singular SDK. If the method implementation just updated SDK state but doesn't use any specific event data then it is expected that there won't be any ReportingMessages, which can be represented by returning an empty collection or `null`

Generally, a good test for whether or not you need a `ReportingMessage` is to check if 1) there is an MParticle object you're using and 2)the MParticle object your using fits any of the `ReportingMessage` static factory methods